### ISSUE TITLE
[lex] Replace "character literal" with references to grammar term character-literal

### ISSUE THREAD

**Issue by zygoloid** [cplusplus/draft#3636]
From CWG discussion of P2029R0.

Remove [lex.ccon] paragraph 1, which defines "character literal" as exactly the same thing as a (grammar term) *character-literal*, and replace all references to "character literal" with references to the grammar term *character-literal*.

Also remove [lex.string] paragraph 1, which defines in plain English the grammar term *string-literal* to be the same thing it already is in the grammar.

**Comment by jensmaurer** [comment]
[lex.icon] and [lex.fcon] have the same problem, fixed by #3650 

Since [lex.string] and [lex.ccon] is being worked on by P2029R0, I'll leave those alone for the time being.
